### DISCUSSION TITLE
Run a command with intent

### DISCUSSION THREAD

**Discussion by mizzunet** [DP-Hridayan/aShellYou#116] [Q&A]
I was wondering if I could run commands on aShell with intent with MacroDroid. 
![Screenshot_20240715-074105_Trebuchet](https://github.com/user-attachments/assets/aa7451d1-08bb-4b23-a662-521c9378fa85)

Thanks

**Comment by DP-Hridayan** [comment]
So macroDroid  is for automation right? What command do you exactly want to
automate?

**Reply by MohamedAliClai** [reply]
I have the same request 
The ADB I want to run for example is 

```
adb -d shell appops set com.catchingnow.tinyclipboardmanager SYSTEM_ALERT_WINDOW allow;

adb -d shell pm grant com.catchingnow.tinyclipboardmanager android.permission.READ_LOGS;

adb -d shell am force-stop com.catchingnow.tinyclipboardmanager;
```

How could I do it via Tasker intent

**Comment by DP-Hridayan** [comment]
There isn't Intent support yet for executing commands.

**Comment by AlexeiCrystal** [comment]
@MohamedAliClai @mizzunet
I don't know if it's up to date. But in Macrodroid, you can send adb/shell commands via Root/Shizuku without the "adb -d shell",
![Screenshot_2025-09-30-20-50-25-356-edit_com arlosoft macrodroid](https://github.com/user-attachments/assets/887bb52f-22c9-4eb8-bc60-4a5f1dc683fa)

**Reply by DP-Hridayan** [reply]
I will have to do some research about this.
